File: WebApi_CapstoneProject/Document Sharing System/Models/DTOs/UserActivityLogDto.cs
namespace DSS.Models.DTOs
{
    public class UserActivityLogDto
    {
        public Guid Id { get; set; }
        public Guid UserId { get; set; }
        public string UserEmail { get; set; } = string.Empty;
        public string UserUsername { get; set; } = string.Empty;
        public string ActivityType { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public DateTime Timestamp { get; set; }
    }

    public class CreateActivityLogDto
    {
        public Guid UserId { get; set; }
        public string ActivityType { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
    }

    public class UserActivitySummaryDto
    {
        public Guid UserId { get; set; }
        public string UserEmail { get; set; } = string.Empty;
        public string UserUsername { get; set; } = string.Empty;
        public DateTime? LastLogin { get; set; }
        public DateTime? LastActivity { get; set; }
        public int TotalActivities { get; set; }
        public int LoginCount { get; set; }
        public int DocumentUploads { get; set; }
        public int DocumentShares { get; set; }
    }
} 

File: Angular_Capstone/DSS/src/app/models/userActivityLogDto.ts
export interface UserActivityLogDto {
  id: string;
  userId: string;
  userEmail: string;
  userUsername: string;
  activityType: string; // "Login", "Logout", "DocumentView", "DocumentShare", "DocumentUpload", etc.
  description: string;
  timestamp: string; // ISO date string
}

export interface UserActivitySummaryDto {
  userId: string;
  userEmail: string;
  totalActivities: number;
  lastActivity: string;
  isActive: boolean; // Based on 30-day activity
  recentActivities: UserActivityLogDto[];
  activityTypeBreakdown: ActivityTypeBreakdown[];
}

export interface ActivityTypeBreakdown {
  activityType: string;
  count: number;
}

export interface CreateActivityLogDto {
  activityType: string;
  description: string;
  userId?: string; // Optional, will be set from token if not provided
}

export interface ApiResponse<T> {
  success: boolean;
  data: T;
  message?: string;
  errors?: string[];
}

export interface ErrorObjectDto {
  errorNumber: number;
  errorMessage: string;
}
